### PR TITLE
Password hash salt logic for signupUser and signinUser

### PR DESCRIPTION
**Pull request by Sriram823**

### CODE REVIEW

**Comment by coltonehrman** [comment]
@Sriram823 please fix the merge conflicts

File: .idea/.gitignore
Comment (coltonehrman) [inline .idea/.gitignore:1]: Please remove this folder from git and add to gitignore so we don't accidently commit again

File: package.json [hunk 1]
```diff
@@ -15,10 +15,13 @@
     "@hookform/resolvers": "^3.3.4",
     "@radix-ui/react-label": "^2.0.2",
     "@radix-ui/react-slot": "^1.0.2",
+    "bcrypt": "^5.1.1",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.0",
+    "dotenv": "^16.4.5",
```
Comment (coltonehrman) [inline package.json:21]: Why was this added?

File: package.json [hunk 2]
```diff
@@ -15,10 +15,13 @@
...
     "lucide-react": "^0.368.0",
-    "mongoose": "^8.3.1",
+    "mongodb": "^6.5.0",
```
Comment (coltonehrman) [inline package.json:23]: Why was this added?

File: package.json (hunk 1 above)
Comment (Sriram823) [reply package.json]: sorry for the unnecessary addition of this  dotenv and mongodb, was trying something but not required regarding the issue and have been removed

**Comment by Sriram823** [comment]
Issues have been resolved, and build is successful. Kindly review the changes.

**Review by Ragudos — CHANGES_REQUESTED** [review]
Please put back the session creation. The reason why you were getting build errors regarding unused variables were because the logic used for those functions got removed.

File: src/lib/User/model.ts
Comment (Ragudos) [inline src/lib/User/model.ts:1]: May I know the reasoning behind storing the salt in the database?

File: src/lib/User/actions.ts
Comment (Ragudos) [inline src/lib/User/actions.ts:1]: May I know the reasoning about adding type assertions? It's advisable to avoid them unless you know exactly that that's the type they're going to be in.

File: src/lib/User/model.ts
Comment (Ragudos) [reply src/lib/User/model.ts:1]: Hmm, I've done some quick search. It's used to compare plaintext to the hashed password. I guess it's fine

File: src/lib/User/actions.ts [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+import bcrypt from "bcrypt";
 
 export async function signupUser(
   data: FormData,
 ): Promise<ZodIssue[] | unknown> {
-  const fullName = data.get("fullName");
-  const email = data.get("email");
-  const password = data.get("password");
-  const confirmPassword = data.get("confirmPassword");
+  const fullName = data.get("fullName") as string;
```
Comment (coltonehrman) [inline src/lib/User/actions.ts:13]: We shouldn't be casting these

File: .gitignore [hunk 4]
```diff
@@ -35,3 +35,7 @@ yarn-error.log*
 # typescript
 *.tsbuildinfo
 next-env.d.ts
+
+#idea
+.idea/
+.idea
```
Comment (coltonehrman) [inline .gitignore:41]: Do we need both these?

File: src/lib/User/model.ts [hunk 5]
```diff
@@ -15,6 +15,7 @@ export interface User extends Document {
   name: string;
   email: string;
   password: string;
+  salt: string;
```
Comment (coltonehrman) [inline src/lib/User/model.ts:18]: Do we need to store the salt?

Comment (Ragudos) [reply src/lib/User/model.ts:18]: His method was generating a salt each time, so I think we do.

**Comment by coltonehrman** [comment]
@all-contributors please add @Sriram823 for code

File: src/lib/User/actions.ts
Comment (Sriram823) [reply src/lib/User/actions.ts:1]: TS was raising errors without them, so i added those.

File: src/lib/User/model.ts (hunk 5 above)
Comment (Sriram823) [reply src/lib/User/model.ts]: Storing the salt alongside the hashed password simplifies the verification process during login attempts.

File: src/lib/User/actions.ts (hunk 3 above)
Comment (Sriram823) [reply src/lib/User/actions.ts]: Oh, i was getting an error prior to casting
